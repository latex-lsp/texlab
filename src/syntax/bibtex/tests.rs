use insta::assert_debug_snapshot;

use super::{parse, SyntaxNode};

fn setup(input: &str) -> SyntaxNode {
    SyntaxNode::new_root(parse(input))
}

#[test]
fn test_preamble() {
    assert_debug_snapshot!(setup(r#"@preamble{"Hello World!"}"#));
}

#[test]
fn test_comment() {
    assert_debug_snapshot!(setup(
        r#"@comment{aksin,
    author       = {Aks{\i}n, {\"O}zge and T{\"u}rkmen, Hayati and Artok, Levent
                    and {\c{C}}etinkaya, Bekir and Ni, Chaoying and
                    B{\"u}y{\"u}kg{\"u}ng{\"o}r, Orhan and {\"O}zkal, Erhan},
    title        = {Effect of immobilization on catalytic characteristics of
                    saturated {Pd-N}-heterocyclic carbenes in {Mizoroki-Heck}
                    reactions},
    journaltitle = jomch,
    date         = 2006,
    volume       = 691,
    number       = 13,
    pages        = {3027-3036},
    indextitle   = {Effect of immobilization on catalytic characteristics},
}"#
    ));
}

#[test]
fn test_entry_colon() {
    assert_debug_snapshot!(setup(
        r#"@article{foo:2019,
    author = {Foo Bar},
    title = {Baz Qux},
    year = {2019},
}"#
    ));
}

#[test]
fn test_biblatex_examples_001() {
    assert_debug_snapshot!(setup(
        r#"@incollection{westfahl:space,
    author       = {Westfahl, Gary},
    title        = {The True Frontier},
    subtitle     = {Confronting and Avoiding the Realities of Space in {American}
                    Science Fiction Films},
    pages        = {55-65},
    crossref     = {westfahl:frontier},
    langid       = {english},
    langidopts   = {variant=american},
    indextitle   = {True Frontier, The},
    annotation   = {A cross-referenced article from a \texttt{collection}. This is
                    an \texttt{incollection} entry with a \texttt{crossref}
                    field. Note the \texttt{subtitle} and \texttt{indextitle}
                    fields},
}

% booktitle and booksubtitle are only needed for BibTeX's less sophisticated
% inheritance set-up to make sure westfahl:space shows correctly.
% With Biber they are not needed.
@collection{westfahl:frontier,
    editor       = {Westfahl, Gary},
    title        = {Space and Beyond},
    date         = 2000,
    subtitle     = {The Frontier Theme in Science Fiction},
    publisher    = {Greenwood},
    location     = {Westport, Conn. and London},
    langid       = {english},
    langidopts   = {variant=american},
    booktitle    = {Space and Beyond},
    booksubtitle = {The Frontier Theme in Science Fiction},
    annotation   = {This is a \texttt{collection} entry. Note the format of the
                    \texttt{location} field as well as the \texttt{subtitle}
                    field},
}"#
    ));
}

#[test]
fn test_biblatex_examples_002() {
    assert_debug_snapshot!(setup(
        r#"@string{jomch   = {J.~Organomet. Chem.}}

@article{aksin,
    author       = {Aks{\i}n, {\"O}zge and T{\"u}rkmen, Hayati and Artok, Levent
                    and {\c{C}}etinkaya, Bekir and Ni, Chaoying and
                    B{\"u}y{\"u}kg{\"u}ng{\"o}r, Orhan and {\"O}zkal, Erhan},
    title        = {Effect of immobilization on catalytic characteristics of
                    saturated {Pd-N}-heterocyclic carbenes in {Mizoroki-Heck}
                    reactions},
    journaltitle = jomch,
    date         = 2006,
    volume       = 691,
    number       = 13,
    pages        = {3027-3036},
    indextitle   = {Effect of immobilization on catalytic characteristics},
}"#
    ));
}

#[test]
fn test_biblatex_examples_003() {
    assert_debug_snapshot!(setup(
        r#"@article{angenendt,
    author       = {Angenendt, Arnold},
    title        = {In Honore Salvatoris~-- Vom Sinn und Unsinn der
                    Patrozinienkunde},
    journaltitle = {Revue d'Histoire Eccl{\'e}siastique},
    date         = 2002,
    volume       = 97,
    pages        = {431--456, 791--823},
    langid       = {german},
    indextitle   = {In Honore Salvatoris},
    shorttitle   = {In Honore Salvatoris},
    annotation   = {A German article in a French journal. Apart from that, a
                    typical \texttt{article} entry. Note the \texttt{indextitle}
                    field},
}"#
    ));
}
