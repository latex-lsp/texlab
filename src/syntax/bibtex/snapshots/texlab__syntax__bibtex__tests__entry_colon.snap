---
source: src/syntax/bibtex/tests.rs
assertion_line: 36
expression: "setup(r#\"@article{foo:2019,\n    author = {Foo Bar},\n    title = {Baz Qux},\n    year = {2019},\n}\"#)"
---
ROOT@0..86
  ENTRY@0..86
    TYPE@0..8 "@article"
    L_DELIM@8..9 "{"
    NAME@9..17 "foo:2019"
    COMMA@17..18 ","
    WHITESPACE@18..23 "\n    "
    FIELD@23..42
      NAME@23..29 "author"
      WHITESPACE@29..30 " "
      EQ@30..31 "="
      WHITESPACE@31..32 " "
      CURLY_GROUP@32..41
        L_CURLY@32..33 "{"
        WORD@33..36 "Foo"
        WHITESPACE@36..37 " "
        WORD@37..40 "Bar"
        R_CURLY@40..41 "}"
      COMMA@41..42 ","
    WHITESPACE@42..47 "\n    "
    FIELD@47..65
      NAME@47..52 "title"
      WHITESPACE@52..53 " "
      EQ@53..54 "="
      WHITESPACE@54..55 " "
      CURLY_GROUP@55..64
        L_CURLY@55..56 "{"
        WORD@56..59 "Baz"
        WHITESPACE@59..60 " "
        WORD@60..63 "Qux"
        R_CURLY@63..64 "}"
      COMMA@64..65 ","
    WHITESPACE@65..70 "\n    "
    FIELD@70..84
      NAME@70..74 "year"
      WHITESPACE@74..75 " "
      EQ@75..76 "="
      WHITESPACE@76..77 " "
      CURLY_GROUP@77..83
        L_CURLY@77..78 "{"
        INTEGER@78..82 "2019"
        R_CURLY@82..83 "}"
      COMMA@83..84 ","
    WHITESPACE@84..85 "\n"
    R_DELIM@85..86 "}"

